packages/workshop-mcp/src/resources.ts

```diff
@@ -168,15 +168,58 @@ async function getExerciseContext({
 	const progress = await getProgress()
 	let stepNumber = 1
 	const playgroundApp = await getPlaygroundApp()
-	invariant(playgroundApp, 'No playground app found')
-	const numbers = extractNumbersAndTypeFromAppNameOrPath(playgroundApp.appName)
-	const isCurrentExercise =
-		exerciseNumber === undefined ||
-		exerciseNumber === Number(numbers?.exerciseNumber)
-	if (exerciseNumber === undefined) {
-		invariant(numbers, 'No numbers found in playground app name')
-		exerciseNumber = Number(numbers.exerciseNumber)
-		stepNumber = Number(numbers.stepNumber)
+	
+	// Handle case where playground is not set
+	let isCurrentExercise = false
+	if (!playgroundApp) {
+		// If no specific exercise is requested and no playground, return helpful message
+		if (exerciseNumber === undefined) {
+			return {
```

Comment:
Can we make what we return here much simpler and just handle this better in the consumer?